services/identity/__tests__/UsersService.spec.ts

```diff
@@ -0,0 +1,168 @@
+import { Sequelize } from "sequelize-typescript"
+import { ModelStatic } from "sequelize/types"
+
+import { User } from "@root/database/models"
+
+import SmsClient from "../SmsClient"
+import TotpGenerator from "../TotpGenerator"
+import _UsersService from "../UsersService"
+
+const MockOtp = {
+  generate: jest.fn(),
+  getExpiryMinutes: jest.fn(),
+  verify: jest.fn(),
+}
+const MockMailer = {
+  sendMail: jest.fn(),
+}
+const MockSmsClient = {
+  sendSms: jest.fn(),
+}
+const MockRepository = {
+  findOne: jest.fn(),
+  update: jest.fn(),
+  findOrCreate: jest.fn(),
+}
+const MockSequelize = {
+  transaction: jest.fn((closure) => closure("transaction")),
+}
+
+const UsersService = new _UsersService({
+  otp: (MockOtp as unknown) as TotpGenerator,
+  mailer: MockMailer,
+  smsClient: (MockSmsClient as unknown) as SmsClient,
+  repository: (MockRepository as unknown) as ModelStatic<User>,
+  sequelize: (MockSequelize as unknown) as Sequelize,
+})
+
+const mockEmail = "someone@tech.gov.sg"
+const mockGithubId = "sudowoodo"
+let DOMAIN_WHITELIST: string | undefined
+
+describe("User Service", () => {
+  beforeAll(async () => {
+    // NOTE: We set the DOMAIN_WHITELIST env var here explicitly
+    // to prevent differing test results between devs
+    DOMAIN_WHITELIST = process.env.DOMAIN_WHITELIST
+    process.env.DOMAIN_WHITELIST = ""
+  })
+
+  afterAll(() => {
+    process.env.DOMAIN_WHITELIST = DOMAIN_WHITELIST
+  })
+
+  afterEach(() => jest.clearAllMocks())
+
+  it("should return the result of calling the findOne method by email on the db model", () => {
+    // Arrange
+    const expected = "user1"
+    MockRepository.findOne.mockResolvedValue(expected)
+
+    // Act
+    const actual = UsersService.findByEmail(mockEmail)
+
+    // Assert
+    expect(actual).resolves.toBe(expected)
+    expect(MockRepository.findOne).toBeCalledWith({
+      where: { email: mockEmail },
+    })
+  })
+
+  it("should return the result of calling the findOne method by githubId on the db model", () => {
+    // Arrange
+    const expected = "user1"
+    MockRepository.findOne.mockResolvedValue(expected)
+
+    // Act
+    const actual = UsersService.findByGitHubId(mockGithubId)
+
+    // Assert
+    expect(actual).resolves.toBe(expected)
+    expect(MockRepository.findOne).toBeCalledWith({
+      where: { githubId: mockGithubId },
+    })
+  })
+
+  it("should return the result of calling the update method by githubId on the db model", async () => {
+    // Arrange
+    const mockUser = "user1"
+
+    // Act
+    await UsersService.updateUserByGitHubId(mockGithubId, mockUser)
+
+    // Assert
+    expect(MockRepository.update).toBeCalledWith(
+      { user: mockUser },
```

Comment:
Should just be `mockUser` instead of `{ user: mockUser }`

Reply:
i'm not sure this is correct ._. changing it to this causes the test to fail sadly

Reply:
oops, i think this issue came up in a downstream branch - fixed [here](https://github.com/isomerpages/isomercms-backend/pull/382/commits/a9df6051d379779aa8763b48b9940909af59c284)